### PR TITLE
Add "Where Sourcegraph is deployed" section

### PR DESCRIPTION
**Pull request by valerybugakov**
It's hard for newbies to find all URLs that host our products. 
This PR has adds a separate section where we can list all the relevant resources.

### CODE REVIEW

File: handbook/engineering/index.md [hunk 1]
```diff
@@ -82,6 +82,12 @@ Sourcegraph has a lot of repositories!
 - [Infrastructure repositories](https://github.com/sourcegraph?utf8=%E2%9C%93&q=repo-type-infrastructure&type=&language=)
 - [Customer infrastructure repositories](https://github.com/sourcegraph?utf8=%E2%9C%93&q=repo-type-infrastructure+repo-type-customer&type=&language=)
 
+## Where Sourcegraph is deployed:
+
+- [Sourcegraph Cloud](https://sourcegraph.com/search)
+- [Sourcegraph About](https://about.sourcegraph.com)
```
Comment (jeanduplessis) [inline handbook/engineering/index.md:88]: @valerybugakov we can probably remove this as it doesn't directly relate to engineering

File: handbook/engineering/index.md [hunk 2]
```diff
@@ -82,6 +82,12 @@ Sourcegraph has a lot of repositories!
 - [Infrastructure repositories](https://github.com/sourcegraph?utf8=%E2%9C%93&q=repo-type-infrastructure&type=&language=)
 - [Customer infrastructure repositories](https://github.com/sourcegraph?utf8=%E2%9C%93&q=repo-type-infrastructure+repo-type-customer&type=&language=)
 
+## Where Sourcegraph is deployed:
```
Comment (jeanduplessis) [inline handbook/engineering/index.md:85]: This heading doesn't really apply to the content. For instance, the Design System is not Sourcegraph. So many something a bit more generic is appropriate here. 

Also, I don't see the QA/Staging instance link here?

**Review by jeanduplessis — CHANGES_REQUESTED** [review]

File: handbook/engineering/index.md (hunk 2 above)
Comment (valerybugakov) [reply handbook/engineering/index.md]: Added more instances and changed title as we discussed in Slack.

**Review by jeanduplessis — APPROVED** [review]

**Comment by jeanduplessis** [comment]
Thanks for putting this together @valerybugakov 👍

**Pull request merged**